application/common/components/adapters/GoogleSheetsIdStore.php

```diff
@@ -0,0 +1,241 @@
+<?php
+namespace Sil\Idp\IdSync\common\components\adapters;
+
+use InvalidArgumentException;
+use Sil\Idp\IdSync\common\components\IdStoreBase;
+use Sil\Idp\IdSync\common\models\User;
+use yii\helpers\Json;
+
+class GoogleSheetsIdStore extends IdStoreBase
+{
+    /**
+     * @var null|string The Application Name to use with Google_Client.
+     */
+    public $applicationName = null;
+    
+    /**
+     * @var null|string The JSON authentication credentials from Google.
+     */
+    public $jsonAuthString = null;
+    
+    /**
+     * @var null|string The Spreadsheet ID.
+     */
+    public $spreadsheetId = null;
+    
+    /**
+     * @var array<string> OAuth Scopes needed for reading/writing sheets.
+     */
+    public $scopes = [\Google_Service_Sheets::SPREADSHEETS];
```

Comment:
since they dont use namespaces the `use` doesnt really shorten it anyway so I just left it this way. Same reason I always do `\Yii::...` when needed rather than a `use Yii; Yii::...`.